packages/reports/addon/helpers/report-action.js

```diff
@@ -1,26 +1,29 @@
 /**
- * Copyright 2017, Yahoo Holdings Inc.
+ * Copyright 2018, Yahoo Holdings Inc.
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  */
-import Ember from 'ember';
-import RouteAction from 'ember-route-action-helper/helpers/route-action';
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+import { get } from '@ember/object';
+import { assert } from '@ember/debug';
 import { ReportActions } from 'navi-reports/services/report-action-dispatcher';
 
-const ROUTE_ACTION = 'onUpdateReport';
+export default Helper.extend({
+  /**
+   * @property {Service} reportActionDispatcher
+   */
+  reportActionDispatcher: service(),
 
-export default RouteAction.extend({
   /**
-   * Returns a closure action that validates the report action
-   * name and sends it out to be dispatched by the report-action-dispatcher
+   * Validates a report action and then dispatches it via the report action dispatcher
    *
-   * @method compute
-   * @override
-   * @params {Array} - The positional arguments to the helper
-   * @returns {Function} - Closure action
+   * @method reportAction
+   * @param {Array} array with the name of the action to dispatch
+   * @returns {Function} Closure action
    */
-  compute([reportAction, ...params]) {
-    let actionName = ReportActions[reportAction];
-    Ember.assert(`The action name "${actionName}" is not a valid report action`, actionName);
-    return this._super([ROUTE_ACTION, actionName, ...params]);
+  compute([action, ...params]) {
+    let actionName = ReportActions[action];
+    assert(`The action name "${action}" is not a valid report action`, actionName);
+    return (() => get(this, 'reportActionDispatcher').dispatch(actionName, ...params));
```

Comment:
I am not sure if this is right? Do we want to create a new dispatcher service for the report-actions?
This service is configured to dispatch actions to update report visualization and request attributes.

This seems like its the same as `update-report-action` except we have removed the middle man `onUpdateReport`.
https://github.com/yahoo/navi/blob/7acd8157bbde0414f1492c0bf80b00d75e17225f/packages/reports/addon/routes/reports/report.js#L223

Reply:
It is essentially the same as `update-report-action` but it just returns a function instead of extending a route-action. The service exists independently of the helper/route-action so I don't see why we'd need a separate one. It works the same way as all the other action-dispatchers that we have.

Reply:
True, but the dispatcher that we have right now dispatches route action. By adding this new consumer we have introduced a normal action that would also be dispatched by the same dispatcher. I feel that the contract for the consumer actions could get confusing if we allow multiple different types of actions to be handled by the same dispatcher, unless ofcourse if we can come up with a contract that is suitable.

Reply:
Basically, calling `update-report-action` returns the `onUpdateReport` route-action after appending a list of actions to dispatch from within that route action. The actions dispatched by `report-action-dispatcher` are still normal actions.

Reply:
Sure, maybe the term route action was incorrect. I mean all the actions currently are always passed a route as the first argument, so they are always associated with a route.

Reply:
I don't see where we're passing a route/routeName anywhere. In the `update-report-action` helper we pass that `onUpdateReport` route-action name, but I don't see anywhere else. The dispatcher takes the same arguments that all the other dispatchers take so I'm not sure how it's any different besides being utilized by the `update-report-action` helper

Reply:
I have the link above where we pass in `this`, which is the route.